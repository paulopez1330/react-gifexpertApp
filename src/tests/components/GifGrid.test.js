import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en <GifGrid/>', () => {
    
    const category = 'mapa'
    // const setCategories = jest.fn();
    // let wrapper = shallow( <GifGrid category = { category } /> );

    beforeEach( () => {
        jest.clearAllMocks();
       /* 
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false,
        });

        wrapper = shallow( <GifGrid category = { category } /> );
        */
    });

    test('debe de mostrar componente correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false,
        });
        
        const wrapper = shallow( <GifGrid category = { category } /> )
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe mostrar items cuando se carga imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'AB',
            url: 'https:localhosts/coaulsd',
            title: 'pureba',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow( <GifGrid category = { category } /> )
        
        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );        
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    })

})