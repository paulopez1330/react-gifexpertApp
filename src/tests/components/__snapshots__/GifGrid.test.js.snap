// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> debe de mostrar componente correctamente 1`] = `
<Fragment>
  <h3>
     
    mapa
     
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
